### ISSUE TITLE
`KDOC_WITHOUT_THROWS_TAG` add `@throws` annotation when throw inside try catch

### ISSUE THREAD

**Issue by benjdero** [saveourtool/diktat#1718]
## Describe the bug
I sometime use `throw` inside a `try catch` when I feel like it makes sense, for example to do extra verifications on user input (see example below).

## Expected behavior
Do nothing

## Observed behavior
`KDOC_WITHOUT_THROWS_TAG` rule add a `@throws` annotation to the function, but it's wrong, this function does not throw anything as it is already caught by the surrounding try catch

## Steps to Reproduce
```
/**
 * @throws NumberFormatException
 */
fun parseInputNumber(onSuccess: (number: Int) -> Unit, onFailure: () -> Unit) {
    try {
        val input: Int = binding.inputEditText.text.toString().toInt()
        if (input < 0)
            throw NumberFormatException()

        onSuccess(input)
    } catch (e: NumberFormatException) {
        onFailure()
    }
}
```

## Environment information
* diktat version: 1.2.5
* build tool (maven/gradle): Gradle 8.2.1 & AGP 8.1.0
* how is diktat run (CLI, plugin, etc.): Plugin
* kotlin version: 1.9.0
* operating system: Ubuntu 22.04
* link to a project (if your project is public): -

**Comment by akuleshov7** [comment]
Hey! thank you for this report! Sounds reasonable, looks like we implemented it in a very-very easy way.
